### PR TITLE
schema/schemaspec/hcl: fix list of ref encode

### PR DESCRIPTION
**Pull request by yonidavidson**

### CODE REVIEW

File: schema/schemaspec/schemahcl/hcl_test.go [hunk 1]
```diff
@@ -109,17 +110,18 @@ show "seinfeld" {
 func ExampleMarshal() {
 	type (
 		Point struct {
-			ID string `spec:",name"`
-			X  int    `spec:"x"`
-			Y  int    `spec:"y"`
+			ID string            `spec:",name"`
+			X  int               `spec:"x"`
+			Y  int               `spec:"y"`
```
Comment (rotemtam) [inline schema/schemaspec/schemahcl/hcl_test.go:115]: Please leave the example test as is (it doesn't make sense for a point to have a Z value that is a ref).

Instead add a specific test in https://github.com/ariga/atlas/blob/master/schema/schemaspec/schemahcl/context_test.go

File: schema/schemaspec/schemahcl/hcl.go [hunk 2]
```diff
@@ -238,9 +238,16 @@ func writeAttr(attr *schemaspec.Attr, body *hclwrite.Body) error {
 	case *schemaspec.ListValue:
 		lst := make([]string, 0, len(v.V))
 		for _, item := range v.V {
-			val, err := schemaspec.StrVal(item)
-			if err != nil {
-				return err
+			var val string
+			var err error
+			switch v := item.(type) {
+			case *schemaspec.Ref:
```
Comment (rotemtam) [inline schema/schemaspec/schemahcl/hcl.go:244]: Can we use an if else (or if-return) block (with type assertion) if we only have 2 cases ?

File: schema/schemaspec/schemahcl/hcl.go [hunk 3]
```diff
@@ -238,9 +238,16 @@ func writeAttr(attr *schemaspec.Attr, body *hclwrite.Body) error {
 	case *schemaspec.ListValue:
 		lst := make([]string, 0, len(v.V))
 		for _, item := range v.V {
-			val, err := schemaspec.StrVal(item)
-			if err != nil {
-				return err
+			var val string
+			var err error
```
Comment (a8m) [inline schema/schemaspec/schemahcl/hcl.go:242]:
```suggestion
			var (
				v string
				err error
			)
```

See Rotem's comment, but group vars into one block.

File: schema/schemaspec/schemahcl/hcl.go [hunk 4]
```diff
@@ -238,9 +238,18 @@ func writeAttr(attr *schemaspec.Attr, body *hclwrite.Body) error {
...
+			)
+			v, ok := item.(*schemaspec.Ref)
+			if ok {
+				val = strings.ReplaceAll(v.V, "$", "")
+			} else {
+				val, err = schemaspec.StrVal(item)
+				if err != nil {
+					return err
+				}
 			}
```
Comment (a8m) [inline schema/schemaspec/schemahcl/hcl.go:253]:
```suggestion
			if v, ok := item.(*schemaspec.Ref); ok {
				val = strings.ReplaceAll(v.V, "$", "")
			} else if val, err = schemaspec.StrVal(item); err != nil {
				return err
			}
```

File: schema/schemaspec/schemahcl/context_test.go [hunk 5]
```diff
@@ -301,3 +301,30 @@ person "jane" {
 	}
 	require.EqualValues(t, jane, test.People[1])
 }
+
+func TestSchemaRefParse(t *testing.T) {
+	type (
+		Point struct {
+			Z []*schemaspec.Ref `spec:"z"`
+		}
+	)
```
Comment (a8m) [inline schema/schemaspec/schemahcl/context_test.go:310]:
```suggestion
	type Point struct {
		Z []*schemaspec.Ref `spec:"z"`
	}
```

**Review by a8m — APPROVED** [review]

**Comment by rotemtam** [comment]
Just realized we both fixed the same bug
https://github.com/ariga/atlas/pull/219/files

**Pull request merged**